### ISSUE TITLE
PS1 should not be set for non-interactive shells

### ISSUE THREAD

**Issue by traylenator** [conda/conda#9163]
* conda-4.6.13-1
* CentOS 7
* Actually the EPEL packages but the profile.d scripts are the same as the conda generated ones.

## Current Behavior

PS1 (bash) or promt (CSH) are always set by `conda.(c).sh` These scripts contain:

* `sh`: `PS1=`
* `csh`: `set prompt=""`

as a fall back always.

## Impact
As per the example.
https://www.tldp.org/LDP/abs/html/intandnonint.html
It is very common to use `PS1` to determine if a session is interactive or not.
In our particular case another profile.d script was doing exactly this which caused some output to be produced and so break scp.

In other words I think PS1 (or prompt) should never set on a non interactive shell.

The other test `tty -s` is a good test to see if a shell is interactive or not and PS1 (or prompt) should not be set in this case.
